### DISCUSSION TITLE
Welcome to Rust Shop!

### DISCUSSION THREAD

**Discussion by dpc** [rustshop/rustshop#1] [Project announcements]
Rust Shop is an SWE Action Role Playing Game. As the founder and self-appointed temporary CTO, I am happy to invite everyone to participate.

In this game, we pretend to be a part of a startup set to disrupt an unspecified industry, by building stuff in Rust. The goal might be fake, but the underlying intentions are real.

Under the pretense of just having fun building stuff, without worrying too much about investors, revenue, market fit, and other boring stuff like that, we can exercise the Rust language and its ecosystem, shave any yaks we want, and ultimately **build an awesome starting point template for bootstrapping Rust-based software companies**.

The whole thing should be easily forkable - and that in the real-world sense, not only git project one. By utilizing Infrastructure As A Code methodologies, and any other necessary means, it should allow bootstraping a small but real-life tech shop with minimum effort. Ideally, any developer should be able to clone this repo locally and by following a relatively short instruction get a usable software system for a small microservices-based web business. This might include creating an AWS account, github access tokens etc. , but should otherwise be highly automated.

Reproducibility has an additional benefit: if it is easy to use this project as a template for a real business, it will be even easier to build another fake one. If the goals, opinions, or social aspects of the group ever become an issue the project can take multiple routes, with each fork setting its own goals and methods. Everyone deserves to be a CTO of their new fake startup if so they desire, or to spend free time working for the fake startup they like best.

To make the game more enjoyable and the results usable, we will need to be mindful about acting out real business constraints and dynamics. We can start small and easy, but as the project evolves we will want to include ever-increasing aspects of production-grade solutions: security, reliability, performance, team organization, and so on.

I hope this is a good entry-level explanation and invitation. If you are interested and/or if you have any questions, feel free to post on any "Meta" category "Discussions" .

**Comment by Michaelin007** [comment]
Nice idea
